examples/08_refetch/src/Item.tsx

```diff
@@ -0,0 +1,26 @@
+import React, { useState } from 'react';
+
+import { store, mutate } from './fetchFuncs';
+import DisplayData from './DisplayData';
+
+type Props = {
+  initialId: string;
+};
+
+const Item: React.FC<Props> = ({ initialId }) => {
+  const [text, setText] = useState(initialId);
+  const result = store.get(initialId);
+  const update = (first_name: string) => {
+    mutate(initialId, first_name).then(() => {
+      store.refetch(initialId);
```

Comment:
If you change this to
```js
store.evict(initialId);
store.prefetch(initialId);
```
doesn't it work?

Reply:
Yeah, definitely, it does. That's how I do it now outside the library. I figured `refetch` would combine those two operations.

Reply:
Okay. That was my biggest concern.
Now, we are on the same page.
A question then comes to if we need to expose `refetch` explicitly.
It doesn't work like ApolloClient's refetch anyway, so wouldn't it be confusing?

Reply:
I don't think it would be confusing. It eliminates a pain point for me, and for others, too, I imagine. I have to create a refetch for all of my queries if there is no method in the library. 

So it saves me one line of repetition on half a dozen queries. Maybe a greater improvement would be to have it not throw a promise...

Reply:
Okay, let me read the example again.

Reply:
Avoiding repetition makes sense, but then do you accept this instead?
```js
  store.prefetch(id, true); // force=true meaning ignore cache
```

---
My question is slightly different: Because `refetch()` doesn't trigger re-render, the app code does need something instead. (that I mean by confusing.)

Reply:
Ah, interesting. I thought throwing a `Promise` would re-render the component?

Reply:
I think using a second argument is more confusing for me personally than an explicit method. It's not clear what `true` refers to here. The name of the method adds to the confusion, since I picture "prefetch" to be used at a specific point in the component lifecycle.

Reply:
Throwing a promise in render will suspend and render again, but throwing it outside render does nothing.
No, "prefetch" should be called outside react render cycle. It's best to be called in event handler.

Yeah, this is a deep issue, it's not just like adding a new api. It's how we deal with Suspense in general.
(Let's move this discussion out of this thread.)